src/main/java/org/opentripplanner/routing/graphfinder/PlaceFinderTraverseVisitor.java

```diff
@@ -176,8 +179,22 @@ private void handleStop(RegularStop stop, double distance) {
       !seenStops.contains(stop.getId()) &&
       (filterByModes == null || stopHasPatternsWithMode(stop, filterByModes))
     ) {
-      placesFound.add(new PlaceAtDistance(stop, distance));
-      seenStops.add(stop.getId());
+      if (
+        includeStopsAndStations &&
+        stop.getParentStation() != null &&
+        !seenStops.contains(stop.getParentStation().getId()) &&
+        !seenStops.contains(stop.getId())
```

Comment:
Isn't this check unnecessary because if the stop has a parent station, we add the parent stations id to the `seenStops`, not the stop's.